src/greyVest/ExpandableTagsInput.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import F from 'futil-js'
+import _ from 'lodash/fp'
+import { observer } from 'mobx-react'
+import { Tag as DefaultTag, Flex } from '.'
+
+let isValidInput = (tag, tags) => !_.isEmpty(tag) && !_.includes(tag, tags)
+
+// We're only using withState to preserve the state between renders, since
+// inject doesn't do that for us.
+let ExpandableTagsInput = ({
+  tags,
+  addTag,
+  removeTag,
+  submit = _.noop,
+  tagStyle,
+  placeholder = 'Search...',
+  splitCommas,
+  style,
+  onBlur = _.noop,
+  onInputChange = _.noop,
+  onTagClick = _.noop,
+  Tag = DefaultTag,
+  collapse,
+  children,
+  ...props
+}) => {
+  let isOneLine = F.view(collapse)
+  addTag = splitCommas
+    ? _.flow(
+        _.split(','),
+        _.invokeMap('trim'),
+        _.compact,
+        _.uniq,
+        _.difference(_, tags),
+        _.map(addTag)
+      )
+    : _.flow(
+        _.trim,
+        addTag
+      )
+  let [currentInput, setCurrentInput] = React.useState('')
+  return (
+    <div
+      className={`tags-input ${isOneLine ? 'tags-input-one-line' : ''}`}
+      style={style}
+    >
+      <span className="tags-input-container" columns="1fr auto" gap="8px 4px">
+        {!(isOneLine && tags.length) && (
+          <input
+            style={{ flex: 1, border: 0 }}
+            onChange={e => {
+              setCurrentInput(e.target.value)
+              onInputChange()
+            }}
+            onBlur={() => {
+              if (isValidInput(currentInput, tags)) {
+                addTag(currentInput)
+                setCurrentInput('')
+                onBlur()
+              }
+            }}
+            onKeyDown={e => {
```

Comment:
there is a ton of overlap here with the regular tags input - let's try to at least factor out the common stuff (e.g. the stuff that manages the tags themselves)